x/crosschain/keeper/v2_zevm_inbound.go

```diff
@@ -183,7 +181,7 @@ func (k Keeper) newWithdrawalInbound(
 		return nil, errors.Wrapf(err, "cannot encode address %v", event.Receiver)
 	}
 
-	gasLimit := event.GasLimit.Uint64()
+	gasLimit := event.CallOptions.GasLimit.Uint64()
```

Comment:
There will be a delay between:
- processing upgrade that will use this new logic
- upgrading the gateway contract to emit the event with callOptions

In practice we should upgrade immediately after the chain upgrade, and pause cctx in between but we should still define the logic handling the event when the contract is not upgrade yet.

Do we have some mechanism to for example reject the tx when the event is not up to date (so not new cctx before contract upgraded)